key-share/src/lib.rs

```diff
@@ -206,15 +265,47 @@ fn validate_vss_key_share<E: Curve>(
     Ok(())
 }
 
-fn validate_non_vss_key_share<E: Curve>(
-    key_share: &DirtyCoreKeyShare<E>,
+fn validate_non_vss_key_info<E: Curve>(
+    shared_public_key: Point<E>,
+    public_shares: &[Point<E>],
 ) -> Result<(), InvalidCoreShare> {
-    if key_share.shared_public_key != key_share.public_shares.iter().sum::<Point<E>>() {
+    let n: u16 = public_shares
+        .len()
+        .try_into()
+        .map_err(|_| InvalidShareReason::NOverflowsU16)?;
+    if n < 2 {
+        return Err(InvalidShareReason::TooFewParties.into());
+    }
+    if shared_public_key != public_shares.iter().sum::<Point<E>>() {
         return Err(InvalidShareReason::SharesDontMatchPublicKey.into());
     }
     Ok(())
 }
 
+impl<E: Curve> DirtyKeyInfo<E> {
+    /// Returns share index of j-th signer
+    ///
+    /// * For additive shares, share id is `j+1`
+    /// * For VSS-shares, share id is scalar $I_j$ such that $x_j = F(I_j)$ where
+    ///   $F(x)$ is polynomial co-shared by the signers
+    ///
+    /// Note: if you have no idea what it is, probably you don't need it.
+    pub fn share_id(&self, j: u16) -> Option<NonZero<Scalar<E>>> {
+        let Some(vss_setup) = self.vss_setup.as_ref() else {
+            return if usize::from(j) < self.public_shares.len() {
+                #[allow(clippy::expect_used)]
+                Some(
+                    NonZero::from_scalar(Scalar::one() + Scalar::from(j))
+                        .expect("1 + i_u16 is guaranteed to be nonzero"),
+                )
```

Comment:
Maybe return None on error case as well? To not pollute the resulting library with strings

Reply:
I can do `unwrap` instead

Reply:
I think it's better to panic when happens something what we didn't expect to happen